src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,199 @@
+     *
+     * @return the corresponding taskList.
+     */
+    public ArrayList<Task> getTasksList(){
+        return tasksList;
+    }
+
+    /**
+     * Print everything in the taskList.
+     */
+    public static void printList() {
+        int count = 1;
+        for(Task task : tasksList){
+            System.out.print(count + ". ");
+            System.out.println(task);
+            count++;
+        }
+    }
+
+    /**
+     * To find corresponding tasks which include the required keyword.
+     *
+     * @param inputLine user input command to show the required keyword.
+     */
+    public static void findTasks(String inputLine) {
+        String findLine = inputLine.substring(5);
+        ArrayList<Task> findTasks = (ArrayList<Task>) tasksList.stream()
+                .filter((t) -> t.getDescription().contains(findLine))
+                .collect(toList());
+        System.out.println("Here are the matching tasks in your list:");
+        int count = 1;
+        for(Task task : findTasks){
+            System.out.print(count + ". ");
+            System.out.println(task);
+            count++;
+        }
+        if(count==1){
+            System.out.println("no matching result");
+        }
+    }
+
+    /**
+     * To add a todo task into the taskList.
+     *
+     * @param inputLine User input command showing the description of the todo task.
+     * @throws DukeException Throw the DukeException.
+     */
+    public static void addTodoTask(String inputLine) throws DukeException{
+        String todoDescription;
+        todoDescription = inputLine.substring(5);
+        Task task = new Todo(todoDescription);
+        //tasks[countThings] = task;
+        tasksList.add(task);
+        Ui.printTask(task);
+
+    }
+
+    /**
+     * To add a deadline task into the taskList.
+     *
+     * @param inputLine User input command showing the description and deadline of the task.
+     * @throws DukeException Throw the DukeException.
+     */
+    public static void addDeadlineTask(String inputLine) throws DukeException{
+        String deadlineDescription;
+        String deadlineByDate;
+        int getIndex;
+        getIndex = inputLine.indexOf("/");
+        deadlineDescription = inputLine.substring(9,getIndex-1);
+        deadlineByDate = inputLine.substring(getIndex+4);
+        Task task = new Deadline(deadlineDescription, deadlineByDate);
+        //tasks[countThings] = task;
+        tasksList.add(task);
+        Ui.printTask(task);
+
+    }
+
+    /**
+     * To add an event to the taskList.
+     *
+     * @param inputLine User input command showing the description and time of the event.
+     * @throws DukeException Throw the DukeException.
+     */
+    public static void addEventTask(String inputLine) throws DukeException{
+        String eventDescription;
+        String eventAtDate;
+        int getIndex;
+        getIndex = inputLine.indexOf("/");
+        eventDescription = inputLine.substring(6,getIndex-1);
```

Comment:
maybe magic number here as well. you should double-check it across you code :+1: